From 136077b40cdabab83aec6833bf3895d4305d30a3 Mon Sep 17 00:00:00 2001
From: Vincas Dargis <vindrg@gmail.com>
Date: Tue, 14 Aug 2018 21:34:59 +0300
Subject: apparmor: fix printing to file

AppArmor denies writing to .ps files and dealing with temporaries needed
for "Print to file" functionality. Add .ps to the allowed extensions
list, also giving access to rm executable together with writing
printing-related temporary file.

Change-Id: I415e1401878bff6459f42162d4e8b517261b9cec
Reviewed-on: https://gerrit.libreoffice.org/59114
Tested-by: Jenkins
Reviewed-by: Rene Engelhard <rene@debian.org>
---
 sysui/desktop/apparmor/program.soffice.bin | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sysui/desktop/apparmor/program.soffice.bin b/sysui/desktop/apparmor/program.soffice.bin
index 4525229..90d663a 100644
--- a/sysui/desktop/apparmor/program.soffice.bin
+++ b/sysui/desktop/apparmor/program.soffice.bin
@@ -37,6 +37,8 @@
 @{libreoffice_ext} += {,x,X}[hH][tT][mM]{,l,L}
 #.epub
 @{libreoffice_ext} += [eE][pP][uU][bB]
+#.ps (printing to file)
+@{libreoffice_ext} += [pP][sS]
 
 #Images
 @{libreoffice_ext} += [jJ][pP][gG]
@@ -117,6 +119,7 @@ profile libreoffice-soffice INSTDIR-program/soffice.bin {
   owner @{HOME}/.cache/fontconfig/**    rw,
   owner @{HOME}/.config/gtk-???/bookmarks r,  #Make bookmarks work
   owner @{HOME}/.recently-used          rwk,
+  owner /tmp/psp[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]* rw, #/tmp/psp1534203998 (printing to file)
 
   owner /{,var/}run/user/*/dconf/user   rw,
   owner @{HOME}/.config/dconf/user      r,
@@ -131,6 +134,7 @@ profile libreoffice-soffice INSTDIR-program/soffice.bin {
   /{usr/,}bin/sh                        rmix,
   /{usr/,}bin/bash                      rmix,
   /{usr/,}bin/dash                      rmix,
+  /{usr/,}bin/rm                        rmix, #deleting /tmp/psp1534203998 (printing to file)
   /usr/bin/bluetooth-sendto             rmPUx,
   /usr/bin/lpr                          rmPUx,
   /usr/bin/paperconf                    rmix,
-- 
cgit v1.1

